src/components/main/Folder.vue

```diff
@@ -0,0 +1,262 @@
+import loggerMixin from "@/mixins/loggerMixin";
+
+export default {
+  data() {
+    return {
+      editedItem: new FormData(),
+      dialog: false,
+      dialogDelete: false,
+      loading: false,
+      threshold: 10,
+      errorMessage: "",
+      editedIndex: -1,
+      documentsUpload: [],
+      headers: [
+        { text: "Nama Dokumen", value: "documentOriginalName" },
+        { text: "Tanggal Diunggah", value: "updatedAt" },
+        { text: "Aksi", value: "actions", sortable: false },
+      ],
+      rules: {
+        required: (val) => val.length > 0 || "Harus diisi",
+        numeric: (val) => /[0-9]/.test(val) || "Masukkan hanya angka",
+        percent: (val) => (val <= 100 && val >= 0) || "Input 0-100",
+      },
+    };
+  },
+  watch: {
+    dialog(val) {
+      val || this.close();
+    },
+    dialogDelete(val) {
+      val || this.closeDelete();
+    },
+  },
+  computed: {
+    folder() {
+      return this.$route.params.item;
+    },
+    docs() {
+      return this.$store.getters["document/documentData"];
+    },
+  },
+  methods: {
+    async initialize() {
+      await this.getDocument();
+    },
+    async getDocument() {
+      this.loading = true;
+      try {
+        let response = await this.$store.dispatch(
+          "document/getDocuments",
+          this.folder._id
+        );
+        if (response.status === 200) {
+          this.errorMessage = "";
+          this.loading = false;
+        }
+      } catch (error) {
+        let message = this.decryptError(error);
+        this.loading = false;
+        EventBus.$emit("onShowSnackbar", message);
+      }
+    },
+    async upload() {
+      this.loading = true;
+      let formData = new FormData();
+      formData.set("creatorId", this.folder.creatorId);
+      formData.set("folderId", this.folder._id);
+      this.documentsUpload.forEach((item) => {
+        formData.append("docs", item);
+      });
+      try {
+        let response = await this.$store.dispatch("document/upload", formData);
+        if (response.status === 200) {
+          this.errorMessage = "";
+        }
+      } catch (error) {
+        let message = this.decryptError(error);
+        EventBus.$emit("onShowSnackbar", message);
+      }
+      this.loading = false;
+      this.documentsUpload = [];
+      this.close();
+    },
+    deleteItem(item) {
+      this.editedIndex = this.docs.indexOf(item);
+      this.editedItem = Object.assign({}, item);
+      this.dialogDelete = true;
+    },
+    async deleteItemConfirm() {
+      console.log("delete on progress");
+      console.log(this.editedItem);
+      // try {
+      //   let response = await this.$store.dispatch(
+      //     "document/delete",
+      //     this.editedItem
+      //   );
+      //   if (response.status === 200) {
+      //     this.initialize();
+      //   }
+      // } catch (error) {
+      //   let message = this.decryptError(error);
+      //   EventBus.$emit("onShowSnackbar", message);
+      // }
```

Comment:
function ini blm used kan ya
mintol dihapus dulu aja.
kebiasaan baiknya
unused code seminimal mungkin

Reply:
apakah setelah aku commit revisinya harus di squash lagi? lalu merge request lagi?

Reply:
ndusah buat baru
lgsung push aja di branch ini
ntar PR ini update sendiri
kalo squash, boleh by local, atau by githubnya.
kalau by github, itu ada dropdown di button Merge Pull Request
nanti milih Squash and Merge